### PR TITLE
Fix Network and Url models and refactor code

### PR DESCRIPTION
**Pull request by elyousfi5**

### CODE REVIEW

File: src/ostorlab/runtimes/local/models/models.py [hunk 1]
```diff
@@ -705,49 +705,124 @@ def create(bundle_id: str = "", application_name: str = "") -> "IosStore":
             return asset
 
 
+class Link(Base):
+    __tablename__ = "link"
+    id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True)
+    url = sqlalchemy.Column(sqlalchemy.String(4096))
+    method = sqlalchemy.Column(sqlalchemy.String(8))
+    url_asset_id = sqlalchemy.Column(
```
Comment (3asm) [inline src/ostorlab/runtimes/local/models/models.py:713]: Urls. Means the other class needs to be changed to Urls with an s.

File: src/ostorlab/runtimes/local/models/models.py [hunk 2]
```diff
@@ -705,49 +705,124 @@ def create(bundle_id: str = "", application_name: str = "") -> "IosStore":
...
     )
-    links = sqlalchemy.Column(sqlalchemy.String(1024))
 
     __mapper_args__ = {
         "polymorphic_identity": "urls",
     }
 
     @staticmethod
-    def create(links: List[str]) -> "Url":
+    def create(links: List[dict]) -> "Url":
```
Comment (3asm) [inline src/ostorlab/runtimes/local/models/models.py:747]: Please type the dict.

File: src/ostorlab/runtimes/local/models/models.py [hunk 3]
```diff
@@ -705,49 +705,124 @@ def create(bundle_id: str = "", application_name: str = "") -> "IosStore":
...
+            url_id: The URL id.
+        """
+        with Database() as session:
+            session.query(Url).filter_by(id=url_id).delete()
+            session.query(Link).filter_by(url_asset_id=url_id).delete()
             session.commit()
-            return asset
+
+
+class IP(Base):
```
Comment (3asm) [inline src/ostorlab/runtimes/local/models/models.py:786]:
```suggestion
class IPRange(Base):
```

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/runtimes/local/models/models.py [hunk 4]
```diff
@@ -835,50 +835,128 @@ def create(
...
     }
 
     @staticmethod
-    def create(links: List[str], scan_id: Optional[int] = None) -> "Url":
+    def create(links: List[dict[str, str]], scan_id: Optional[int] = None) -> "Urls":
         """Persist the URL information in the database.
 
         Args:
-            links: list of the target URLs.
+            links: list of the target URLs with methods.
```
Comment (benyissa) [inline src/ostorlab/runtimes/local/models/models.py:881]: Could you please add scan_id to the Args section in the docstring too?

File: src/ostorlab/runtimes/local/models/models.py [hunk 5]
```diff
@@ -657,20 +657,20 @@ def create_from_assets_definition(
         """Create the assets from the asset definition.
 
         Args:
-            scan: The scan object.
             assets: The list of assets to create.
+            scan_id: The scan id.
         """
-        networks: List[str] = []
-        links: List[str] = []
+        networks = []
```
Comment (amine3) [inline src/ostorlab/runtimes/local/models/models.py:663]: Why remove the typing?

**Review by amine3 — DISMISSED** [review]

File: src/ostorlab/runtimes/local/models/models.py [hunk 6]
```diff
@@ -657,20 +657,20 @@ def create_from_assets_definition(
         """Create the assets from the asset definition.
 
         Args:
-            scan: The scan object.
             assets: The list of assets to create.
+            scan_id: The scan id.
         """
-        networks: List[str] = []
-        links: List[str] = []
+        networks: List[Dict[str, Union[str, int]]] = []
```
Comment (deadly-panda) [inline src/ostorlab/runtimes/local/models/models.py:663]: the int here is for the mask? the proto define it as a string..

Comment (deadly-panda) [inline src/ostorlab/runtimes/local/models/models.py:663]: the `int` here is for the mask? the proto & asset instance both define it as a string..

File: src/ostorlab/runtimes/local/models/models.py [hunk 7]
```diff
@@ -835,63 +839,163 @@ def create(
...
-            )
-            session.add(asset)
+            network_instance = Network(scan_id=scan_id)
+            session.add(network_instance)
+            session.commit()
+
+            network_items = [
+                IPRange(
+                    host=network.get("host"),
+                    mask=network.get("mask", 32),
```
[the remark this answers is not in the document]
Comment (elyousfi5) [reply src/ostorlab/runtimes/local/models/models.py:982]: nice catch

File: src/ostorlab/runtimes/local/models/models.py (hunk 6 above)
Comment (elyousfi5) [reply src/ostorlab/runtimes/local/models/models.py:663]: true, but it doesnt create any conflict here !!

**Review by 3asm — APPROVED** [review]

**Pull request merged**